llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -5022,6 +5022,12 @@ InstCombinerImpl::foldICmpWithMinMaxImpl(Instruction &I,
     std::swap(CmpXZ, CmpYZ);
   }
 
+  auto FoldIntoCmpYZ = [&]() -> Instruction * {
+    if (CmpYZ.has_value())
+      return replaceInstUsesWith(I, ConstantInt::getBool(I.getType(), *CmpYZ));
+    return ICmpInst::Create(Instruction::ICmp, Pred, Y, Z);
+  };
```

Comment:
This lambda is only used once.
Imo its clearer to just leave this logic inlined at its callsite.

Reply:
It is also used in
https://github.com/llvm/llvm-project/blob/07151f0241d3f893cb36eb2dbc395d4098f74a87/llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp#L5079
and
https://github.com/llvm/llvm-project/blob/07151f0241d3f893cb36eb2dbc395d4098f74a87/llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp#L5088